tests/meson.build

```diff
@@ -0,0 +1,46 @@
+runtest = files(join_paths(meson.project_source_root(), 'runtest.sh'))
+
+unittest_multi = executable('unittest_multi', src_inih, 'unittest.c', c_args : ['-Wall'])
+test('test_multi', runtest, depends : [unittest_multi], args : [files('baseline_multi.txt'), unittest_multi.full_path()])
+
+unittest_multi_max_line = executable('unittest_multi_max_line', src_inih, 'unittest.c', c_args : ['-Wall', '-DINI_MAX_LINE=20'])
+test('test_multi_max_line', runtest, depends : [unittest_multi_max_line], args : [files('baseline_multi_max_line.txt'), unittest_multi_max_line.full_path()])
```

Comment:
I don't know what's possible with meson, but it looks like there's data structures. Is it possible to write these as something like a table-driven test, and only include the variables (things that are different) in the table data. Maybe something like:

```
tests = [
    ['multi', ['-Wall']],
    ['multi_max_line', ['-Wall', '-DINI_MAX_LINE=20']],
    ['single', ['-Wall', '-DINI_ALLOW_MULTILINE=0']],
    ...
]
for name, c_args in tests:
    test('test_'+name, runtest, ...)
```